### PR TITLE
[KernelGen][Nvidia] Add logcumsumexp operator with Triton kernel

### PR DESCRIPTION
**Pull request by Dingxingdi**
## Summary
Adds Triton kernel support for `logcumsumexp` and `logcumsumexp.out`.

## Testing
- Validated against PyTorch reference on device side
- Accuracy: `pytest tests/test_logcumsumexp.py -x -v --timeout=60`
- Benchmark: `pytest benchmark/test_logcumsumexp.py --level core -s`
- Tested on: Nvidia H20, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_logcumsumexp.py --level core -s` (NVIDIA H20)

| Variant | dtype | Cases | Mean Speedup |
|---|---:|---:|---:|
| logcumsumexp | float16 | 5 | 4.47x |
| logcumsumexp | float32 | 5 | 4.02x |
| logcumsumexp | bfloat16 | 5 | 4.22x |
| **logcumsumexp** | **all** | **15** | **4.269x** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 4.269x | Primary |
| Tianshu | FAIL | PASS | 5.085x | From existing backend summary |
| Muxi | FAIL | PASS | 0.985x | From existing backend summary |
| Ascend | FAIL | N/A | — | From existing backend summary |
| Hygon | PASS | PASS | 7.194x | From existing backend summary |

## Files Changed
- `src/flag_gems/ops/logcumsumexp.py`: Triton kernel implementation
- `tests/test_logcumsumexp.py`: Accuracy tests for `logcumsumexp` and `logcumsumexp.out`
- `benchmark/test_logcumsumexp.py`: Performance benchmarks
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entries (stage: alpha 5.4)

### CODE REVIEW

**Review by w1120029931-bit — COMMENTED** [review]
## FlagGems Automated Review

**Operator:** `logcumsumexp`
**Verdict:** COMMENT

### Summary
- Errors: 0
- Warnings: 3
- Suggestions: 0

File: src/flag_gems/ops/logcumsumexp.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Generated by KernelGen: https://github.com/flagos-ai/KernelGen
+import functools
+import logging
+
+import torch
+import triton
+import triton.language as tl
+from torch._prims_common import is_boolean_dtype, is_integer_dtype
```
Comment (w1120029931-bit) [inline src/flag_gems/ops/logcumsumexp.py]: **Suggestion**: Usage of private torch API: torch._prims_common

**Recommended fix:**
Use public torch APIs instead of private internals.

<sub>Rule: `KERNEL_NO_PRIVATE_API`</sub>

File: src/flag_gems/ops/logcumsumexp.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        dtype = inp.dtype
+        if is_integer_dtype(dtype) or is_boolean_dtype(dtype):
+            dtype = torch.float32
+    if out is None:
+        out = torch.empty_like(inp, dtype=dtype)
+
+    with torch_device_fn.device(inp.device):
+        if K == 1:  # row scan
+            TILE_SIZE = triton.next_power_of_2(N)
+            num_warps = 8 if TILE_SIZE > 2048 else 4
```
Comment (w1120029931-bit) [inline src/flag_gems/ops/logcumsumexp.py:72]: **Suggestion**: Hardcoded hyperparameter without comment: num_warps = 8 if TILE_SIZE > 2048 else 4

**Recommended fix:**
Add a comment explaining why this value was chosen.

<sub>Rule: `HYPERPARAMS`</sub>

File: src/flag_gems/ops/logcumsumexp.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                inp, out, M, N, TILE_SIZE, num_warps=num_warps
+            )
+        else:  # column scan (K > 1)
+            # For K > 1, each of the M*K rows is independent
+            # We scan along dimension N (dim), and K becomes the "batch" dimension
+            # Layout: (M, N, K) -> treat as (M*K, N) with K as batch
+            # Actually for simplicity, let's handle the K > 1 case with a different approach
+            # by processing each (M, K) combination independently
+            TILE_SIZE = triton.next_power_of_2(N)
+            num_warps = 8 if TILE_SIZE > 2048 else 4
```
Comment (w1120029931-bit) [inline src/flag_gems/ops/logcumsumexp.py:84]: **Suggestion**: Hardcoded hyperparameter without comment: num_warps = 8 if TILE_SIZE > 2048 else 4

**Recommended fix:**
Add a comment explaining why this value was chosen.

<sub>Rule: `HYPERPARAMS`</sub>

File: src/flag_gems/ops/logcumsumexp.py (hunk 1 above)
Comment (Dingxingdi) [reply src/flag_gems/ops/logcumsumexp.py]: Done.

File: src/flag_gems/ops/logcumsumexp.py (hunk 2 above)
Comment (Dingxingdi) [reply src/flag_gems/ops/logcumsumexp.py:72]: Done.

File: src/flag_gems/ops/logcumsumexp.py (hunk 3 above)
Comment (Dingxingdi) [reply src/flag_gems/ops/logcumsumexp.py:84]: Done.

**Review by bin913 — APPROVED** [review]
